index.js

```diff
@@ -20,9 +20,9 @@ function loop() {
 }
 
 function advance() {
+  if (this.step >= this.division) this.step = 0;
```

Comment:
Well, I am working under the assumption that the sequence can be empty because that's the way I have been using this. If we init it the way you describe below then it's different.